File: src/domain/entities/messaging/MessageReference.ts
/**
 * WhatsApp message identifier used by reply, reaction and other content that targets
 * an existing message.
 */
export class MessageReference {
  constructor(
    public readonly messageId: string,
    public readonly remoteJid?: string,
    public readonly participantId?: string,
  ) {
    if (!messageId.trim()) {
      throw new Error('MessageReference requires a non-empty messageId.');
    }
  }
}

File: sdks/java/src/main/java/com/digows/whatsappgateway/messaging/MessageReference.java
package com.digows.whatsappgateway.messaging;

public record MessageReference(
  String messageId,
  String remoteJid,
  String participantId
)
{
  public MessageReference
  {
    if (messageId == null || messageId.isBlank())
    {
      throw new IllegalArgumentException("MessageReference requires a non-empty messageId.");
    }
  }
}
